packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js

```diff
@@ -51,31 +59,25 @@ export default function ClassifierWrapper({
       subjectId: subject.id,
       locations: subject.locations
     })
-  }
+  }, [recents?.add, yourStats?.increment])
 
-  function onError(error, errorInfo={}) {
-    logToSentry(error, errorInfo)
-    console.error('Classifier error', error)
-  }
-
-  function onSubjectChange(subject) {
-    const { query } = router
-    const baseURL = `/${query.owner}/${query.project}/classify`
-    if (query.subjectID && subject.id !== query.subjectID) {
+  const onSubjectChange = useCallback((subject) => {
+    const baseURL = `/${ownerSlug}/${projectSlug}/classify`
+    if (subjectID && subject.id !== subjectID) {
       const newSubjectRoute = `${baseURL}/workflow/${workflowID}/subject-set/${subjectSetID}/subject/${subject.id}`
       const href = addQueryParams(newSubjectRoute)
       const as = href
       router.replace(href, as, { shallow: true })
     }
-  }
+  }, [ownerSlug, projectSlug, router?.replace, subjectID])
```

Comment:
This callback also uses the workflow ID and subject set ID props, which are passed down from the page URL.

Reply:
Deciphering Secrets seems to work even though these are missing.